src/asm.rs

```diff
@@ -485,13 +485,21 @@ impl<'a, 'gcc, 'tcx> AsmBuilderMethods<'tcx> for Builder<'a, 'gcc, 'tcx> {
             // TODO(@Commeownist): figure out how to align stack
         }
 
-        // Write results to outputs
+        // Write results to outputs. 
+        //
+        // We need to do this because:
+        //  1. Turning `PlaceRef` into `RValue` is error-prone and has nasty edge cases 
+        //     (especially with current `rustc_backend_ssa` API).
+        //  2. Not every output operand has an `out_place`, and it's required by `add_output_operand`.
+        //
+        // Instead, we generate a temporary output variable for each output operand, and then this loop,
+        // generates `out_place = tmp_var;` assignments if out_place exists, or `(void)tmp_var;` if it doesn't.
         for op in &outputs {
             if let Some(place) = op.out_place {
                 OperandValue::Immediate(op.tmp_var.to_rvalue()).store(self, place);                
+            } else {
+                block.add_eval(None, op.tmp_var.to_rvalue());
```

Comment:
`add_eval()` is sometimes necessary to make something known to the backend, for instance, to execute something that does not return a value (for instance, a call to `printf` where you don't care about the result).

Here, I don't know if such a thing happens. The only thing I could see is telling to the backend the register name, but even then, it might already work without calling `add_eval()` (I believe this is the case, but I might be wrong).

Am I missing something here?

Reply:
Not really. I'm just getting familiar with the code. I saw this thing is used when calling `void` functions, and made a guess, a wrong one. Do you want me to undo?

Reply:
Yes please!